### DISCUSSION TITLE
Feature request - support for OS 3.4.0 - re-enable ssh service

### DISCUSSION THREAD

**Discussion by darxis** [garrynewman/Control4.Jailbreak#31] [Ideas]
Upgrading to OS 3.4.0 disables the terminal (SSH) as stated in the Release Notes, so you can get locked out as the Jailbreak uses SSH to upload the certificates to the controller.

One way to start the SSH service requires already having access to the director via Composer Pro and executing the following Lua commands via any driver Lua execute tab:
```lua
local handle = io.popen("service sshd start")
local out = handle:read("*a")
handle:close()
print(out)
```

Another way to access a terminal would be to connect to the UART serial port via a USB cable (assuming the controller has a UART serial port).

It would be nice if this Jailbreak would automatically re-enable the SSH service for 3.4.0 so it can upload the certificates for the Jailbreak to work. I don't know if it is possible.

Or maybe we should think about another method to Jailbreak? Like finding a way using the original built-in Control4 certificates on the Controller and Composer Pro so that we don't have to generate new ones and upload them to the controller?

**Comment by cleemullins** [comment]
I didn't realize it would be so easy to restart the SSH service. If this was packaged up as a "restart SSH Driver", then you just need a remote dealer to login and install the driver. A button on a room could trigger it, the certs then are installed, and SSH is turned off some time later. 

It's not a perfect workaround, but its... not bad.
